### ISSUE TITLE
Add support for variable_type field on variabld (project and group)

### ISSUE THREAD

**Issue by sudoapt-getclean** [xanzy/go-gitlab#639]
GitLab allows settings group and project variables with type 'file'. GitLab CI creates a temporary file with the contents of the variable while the actual env variable gets set to the path to that file. [Gitlab docs.](https://docs.gitlab.com/ee/ci/variables/#variable-types)

The gitlab API supports file variables for both [groups](https://docs.gitlab.com/ce/api/group_level_variables.html#update-variable) and [projects](https://docs.gitlab.com/ce/api/project_level_variables.html#create-variable).

I would like to use these fields [in the terraform-gitlab provider.](https://github.com/terraform-providers/terraform-provider-gitlab/issues/153)

**Comment by svanharmelen** [comment]
@sudoapt-getclean could you check if PR #653 works for you use case?

**Comment by svanharmelen** [comment]
@sudoapt-getclean can you config if this solves your use case?

**Comment by sudoapt-getclean** [comment]
I'm out camping with the family for 2 more days than I'm gonna pick it up
asap

**Comment by svanharmelen** [comment]
All good, no rush! Enjoy your holidays!

**Comment by svanharmelen** [comment]
@sudoapt-getclean any feedback/update on this one?

**Comment by sudoapt-getclean** [comment]
Hell Yeah!
I just completed an implementation based on this and it is looking very good. 
(it took longer than I wanted because I had to sit down and figure out how to use the code from the local repo instead of github.(GO not being my daily language)).

My only remark is that it might be time to remove `build_variables.go` and `build_variables_test.go` since they have now been completely superseded by the project and group variables.
(i've done it locally so I could submit a pr but it's a trivial change so it won't save you any work, anyway let me know)

I suppose that could break other consumers of the library but I would say at the benefit of forcing them to make it clear what they are actually doing.

thanx for the work and I'm looking forward to v0.20
